README.md

```
@@ -251,6 +251,33 @@ uv run black src/ tests/ && uv run isort src/ tests/
 - [DEVELOPMENT.md](DEVELOPMENT.md): Architecture, API reference, contribution guide
 - [RESEARCH.md](RESEARCH.md): QADI methodology background
 - [SESSIONS.md](SESSIONS.md): Development history
+- [SEMANTIC_OPERATORS_IMPLEMENTATION.md](SEMANTIC_OPERATORS_IMPLEMENTATION.md): Semantic evolution operators guide
+
+## Future Improvements
+
+### Performance Optimizations
```

markdownlint-cli2 (0.17.2)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)